osbs/utils.py

```diff
@@ -256,7 +260,11 @@ def clone_git_repo(git_url, target_dir=None, commit=None, retry_times=GIT_MAX_RE
             # we are using check_output, even though we aren't using
             # the return value, but we will get 'output' in exception
             subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-            repo_commit, repo_depth = reset_git_repo(target_dir, commit, depth)
+            try:
+                repo_commit, repo_depth = reset_git_repo(target_dir, commit, depth)
```

Comment:
I realized that this is used in atomic-reactor. So we have to rework exception to be subclass of `OsbsException`


IMHO we can re-raise the same exception type with bigger context.

```
try:
    ....
except OsbsCommitNotFound as exc:
    raise OsbsCommitNotFound("......".format(branch, exc))
```